code/datums/chatmessage.dm

```diff
@@ -190,23 +218,97 @@
 
 	// Prepare for destruction
 	scheduled_destruction = world.time + (lifespan - CHAT_MESSAGE_EOL_FADE)
-	addtimer(CALLBACK(src, PROC_REF(end_of_life)), lifespan - CHAT_MESSAGE_EOL_FADE, TIMER_UNIQUE|TIMER_OVERRIDE)
 
+	INVOKE_ASYNC(src, PROC_REF(spelling_loop))
+
+/datum/chatmessage/proc/turn_to_styled(string)
+	return {"<span style='font-size:[font_size]pt;font-family:"Pterra";color:[tgt_color];text-shadow:0 0 5px #000,0 0 5px #000,0 0 5px #000,0 0 5px #000;' class='center maptext [_extra_classes != null ? _extra_classes.Join(" ") : ""]' style='color: [tgt_color]'>[string]</span>"} //AAAAAAAAAAAAAAA
+
+/datum/chatmessage/proc/spelling_extra_delays(character)
+	if(character in CHAT_SPELLING_EXCEPTIONS)
+		return
+
+	return CHAT_SPELLING_PUNCTUATION[character] ? CHAT_SPELLING_PUNCTUATION[character] : 0
+
+/datum/chatmessage/proc/spelling_loop()
+	if(QDELETED(src))
+		return
+
+	var/delay = CHAT_SPELLING_DELAY_WITH_EXCLAIMED_MULTIPLIER
+	var/direction = 1
 
+	for(var/letter as anything in remaining_letters)
+		var/extra_delay = spelling_extra_delays(letter)
+		if(isnull(extra_delay))
+			current_string += letter
+			continue
 
+		addtimer(CALLBACK(src, PROC_REF(add_letter), letter, direction, (extra_delay ? FALSE : TRUE)), delay)
+		direction *= -1
+		delay += CHAT_SPELLING_DELAY_WITH_EXCLAIMED_MULTIPLIER + extra_delay
 
+	addtimer(CALLBACK(src, PROC_REF(end_of_life)), delay + 2 SECONDS)
+
+/datum/chatmessage/proc/add_letter(letter, direction, audible)
+	if(QDELETED(src))
+		return
+	if(premature_end)
+		return
+
+	_add_letter(arglist(args))
+
+/datum/chatmessage/proc/_add_letter(letter, direction, audible)
+	current_string += letter
+	message.maptext = MAPTEXT(turn_to_styled(current_string))
+	if(audible && !_extra_classes.Find("emote"))
+		play_toot()
+		do_shift(direction)
```

Comment:
ignoring the toot bit, I suspect you could just do this as an animate chain (since animate can set maptext, just no interpolation is all). That prevents a bunch of timer spam, and would make this code far lighter. lot of string shit but like that's gonna happen either way.